### ISSUE TITLE
VueJS and FastAPI example

### ISSUE THREAD

**Issue by jd-solanki** [r0x0r/pywebview#1175]
### Specification
- pywebview version: na
- operating system: na
- web renderer: na

### Description

Thanks for this library ❤️ 

I want to use this for python based application like raycast. After trying various gui I thigh webview will be faster and here  I am.

Can you please provide example of using vuejs and fastapi to consume this package??

Additionally, I wanted to ask are you going to maintain this in future??

 
### Practicalities

- YES, I am willing to work on this issue myself.

**Comment by jd-solanki** [comment]
Moreover, I wonder will I be able use vue dev server while developing the app or I have to create standalone html,css n js??

This is because I prefer to use nuxtjs if vue is supported.

**Comment by zyugyzarc** [comment]
> I wonder will I be able use vue dev server while developing the app or I have to create standalone html,css n js??
Yes, just set the `url` to the vue-dev-server url when creating the window.

For default vue-cli settings, use:
```python
webview.create_window(url="http://0.0.0.0:8888/")
```

If you want the vue server to be started in the python script, you the `func` argument in `webview.start()`. For example:

```
import os

def webview_init(window):
    os.system("npm run serve") # or use `yarn serve`, or `npx vue-cli-service serve`

window = webview.create_window("Vue example", url="http://0.0.0.0:8080/")
webview.start(webview_init, window)
```

for more, see <https://pywebview.flowrl.com/guide/usage.html#threading-model>

This method is only recommended for development, since distributing this will require the end user to install npm, vue-cli and a lot more dependencies - but since you seem to be using nuxt, I assume you plan to distribute a client application, while keeping the nuxtjs and vue part server-side.

**Comment by jd-solanki** [comment]
I tried giving it a shot but I couldn't able to put things together? Could you please provide more details on what to do and what it will perform under the hood. I'm intrested in learning how things works.

Additionally, Will frontend code render dynamic data using API? If so, will it make app slower?

**Comment by zyugyzarc** [comment]
Here's a simple fastapi example that you could use:

```python
import uvicorn
import webview
from fastapi import FastAPI


app = FastAPI()


@app.get("/")
async def read_root():
    return {"Hello": "World"}


def main(window):
    uvicorn.run(app, host="0.0.0.0", port=8000)

window = webview.create_window('Fastapi example', app, url="http://localhost:8000/")
webview.start(main, window)
```

and for vue, it should be pretty similar to this react example: <https://pywebview.flowrl.com/examples/#react-boilerplate>

see also:
<https://vuejs.org/examples/#hello-world>
<https://testdriven.io/blog/developing-a-single-page-app-with-fastapi-and-vuejs/>
<https://stackoverflow.com/questions/62455652/how-to-serve-static-files-in-fastapi>
<https://pywebview.flowrl.com/guide/usage.html#http-server>

**Comment by jd-solanki** [comment]
Hi @zyugyzarc Thanks for the response.

I'm able to run the app using fastapi server. In your example, we have to make below changes:
```diff
- window = webview.create_window('Fastapi example', app, url="http://localhost:8000/")
+ window = webview.create_window("Fastapi example", server=app, url="http://localhost:8000/")
```

Anyway, I decided to use web server approach where I had to let go of Vue. I'll use fastapi & its template mechanism.

However, What's your suggestions on choosing pattern of pywebview for building app like raycast/alfred. (This is because I have hard time figuring out how can I use python for application logic & vue for frontend)
